.github/workflows/branch-sync.yml

```diff
@@ -13,71 +13,90 @@ permissions:
 jobs:
   create-pull-request:
     runs-on: ubuntu-latest
+    container: ghcr.io/dpc-sdp/bay/ci-builder:6.x
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
         with:
-          fetch-depth: 0
+          ref: 6.x
+
+      - name: Set ownership of the workspace
+        run: chown -R $(id -u):$(id -g) $PWD
 
       - name: Configure git
         run: |
           git config user.name "github-actions[bot]"
           git config user.email "github-actions[bot]@users.noreply.github.com"
 
-      - name: Prepare sync branch
-        shell: bash
+      - name: Create PR
+        id: create-pr
         run: |
-          set -euo pipefail
-
-          git fetch origin
+          set +e
+          output=$(gh pr create --base 7.x --title "Automated PR - merge 6.x into 7.x" --body "This pull request syncs relevant changes from the 6.x branch into the 7.x branch. This PR was automatically generated by the CI workflow." --label "sync" --label "automated" 2>&1)
+          COMMAND_STATUS=$?
```

Comment:
Capture the command status to exclude the response `
pull request create failed: GraphQL: No commits between 7.x and 6.x (createPullRequest)` from failing the job.